lib/tasks/oregon_digital/csv_ingest.rake

```diff
@@ -0,0 +1,124 @@
+#what does the csv look like?
+#  fields in the top line
+#  what about multiple values? pipe as a delimiter
+#  worktype in the first col, eg generic, image ...
+#  filenames in the second col
+#  id in the third col, omit "oregondigital:"
+#  if read_groups is included, this will override the default of public
+#  if admin_set_id is included, it will override admin_set/default
+#args: path to dir containing all files including csv, name of the csv, email
+#called like this: rake oregon_digital:csv_ingest csv=mymetadata.csv filedir=/data email=admin@example.org
+
+WORKTYPE = 0
+FILENAMES = 1
+ID = 2
+SKIPTO = 3
+
+namespace :oregon_digital do
+  desc "ingest works using csv"
+  task csv_ingest: :environment do
+    csv = ENV['csv']
+    filedir = ENV['filedir']
+    email = ENV['email']
+    process(csv, filedir, email)
+  end
+end
+
+def process(csv, filedir, email)
+  user = User.find_by_user_key(email)
+  logger = create_logger
+  csv = File.open("#{filedir}/#{csv}")
+  fields = csv.readline.split("\t")
+  csv.readlines.each do |line|
+    work = process_line(line, fields, email, logger)
+    work.read_groups = ["public"] unless !work.read_groups.empty?
+    work_with_set = set_admin_set(work)
+    uploaded_files = prep_files(line, filedir, user, logger)
+    if !uploaded_files.empty?
+      Hyrax::AttachFilesToWorkJob.perform_later(work_with_set, uploaded_files)
+      logger.info "saved item #{work.id} and queued AttachFilesToWorkJob"
+    else
+      logger.error "Did not attach files to #{work.id}"
+    end
+  end
+  csv.close
+end
+
+def process_line(line, fields, email, logger)
+    vals = line.split("\t")
+    work = send(vals[WORKTYPE])
```

Comment:
Instead of this, you can do

`work = vals[WORKTYPE].humanize.constantize.new` and remove the related methods down below